AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/AWSAuthSignOutTask.swift

```diff
@@ -26,8 +26,17 @@ class AWSAuthSignOutTask: AuthSignOutTask {
 
     func execute() async -> AuthSignOutResult {
         await taskHelper.didStateMachineConfigured()
-        await sendSignOutEvent()
-        return await doSignOut()
+        if  case .configured(let authNState, _) = await authStateMachine.currentState,
+            isValidAuthNStateToStart(authNState) {
+            await sendSignOutEvent()
+            return await doSignOut()
+        } else {
+            let invalidStateError = AuthError.invalidState(
```

Comment:
I believe this error message is tied to federation but is in a generic else block, not ensuring the path is due to federation.

Reply:
Nice catch.. Updated.. :thank-you: